server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -666,7 +667,12 @@ private static <Request extends TransportRequest> TransportRequestHandler<Reques
     ) {
         var threadPool = transportService.getThreadPool();
         var latencies = searchTransportMetrics.getActionLatencies();
-        Map<String, Object> attributes = Map.of(ACTION_ATTRIBUTE_NAME, actionQualifier);
+        Map<String, Object> attributes = Map.of(
+            ACTION_ATTRIBUTE_NAME,
+            actionQualifier,
+            SYSTEM_THREAD_ATTRIBUTE_NAME,
+            ThreadPool.isSystemThreadPool()
```

Comment:
At a quick glance, it seems we're not on the correct thread when this function is invoked. The forking in the search thread pool occurs later, so this should likely be moved to the lambda defined below?

Reply:
🤦 Correct - fixed in ea6a2ae837e87abade62c7df8e651394fc06e750.

Would there be a simple way to test this? Is there a way to ensure that the search is going to be forked to the corresponding executor?

Reply:
Looking at the callers of `instrumentedHandler` it looks like we're not measuring the right thing. Most of the function call referenced in the instrumented handler lambdas are asynchronous and will delegate the actual execution of the search in a different thread. 
See for instance [here](https://github.com/elastic/elasticsearch/blob/97a724e499b6aae5c819f90d0c14a84a746a6ca3/server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java#L526) where we call `SearchService#executeQueryPhase`. However this function is not synchronous so I think we're just measuring the time it takes to move the actual execution on a different thread? cc @javanna